core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,116 +288,3206 @@ def validate(cls, item):
+            cls._validate_post_commit_is_private(item)
+
+        if cls.is_snapshot_metadata_model:
+            cls._validate_commit_type(item)
+            cls._validate_commit_cmds_schema(item)
+
         for func in cls._get_validation_functions():
             func(item)
 
 
-class UserSubscriptionsModelValidator(BaseModelValidator):
-    """Class for validating UserSubscriptionsModels."""
+class ActivityReferencesModelValidator(BaseModelValidator):
+    """Class for validating ActivityReferencesModels."""
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        regex_list = [
+            item + '$' for item in feconf.ALL_ACTIVITY_REFERENCE_LIST_TYPES]
+        regex_string = '|'.join(regex_list)
+        return regex_string
+
+    @classmethod
+    def _get_json_properties_schema(cls, item):
+        activity_references_dict_schema = {
+            'type': 'dict',
+            'properties': [{
+                'name': 'type',
+                'schema': {
+                    'type': 'unicode',
+                },
+            }, {
+                'name': 'id',
+                'schema': {
+                    'type': 'unicode'
+                }
+            }]
+        }
+        activity_references_schema = {
+            'type': 'list',
+            'items': activity_references_dict_schema
+        }
+
+        return {
+            'activity_references': activity_references_schema
+        }
+
+    @classmethod
+    def _get_model_domain_object_instances(cls, item):
+        model_domain_object_instances = []
+
+        try:
+            for obj in item.activity_references:
+                model_domain_object_instances.append(
+                    activity_domain.ActivityReference(obj['type'], obj['id']))
+        except Exception as e:
+            cls.errors['fetch properties'] = (
+                'Model id %s: Model properties cannot be fetched completely '
+                'with the error %s') % (item.id, e)
+            return []
+
+        return model_domain_object_instances
+
+    @classmethod
+    def _get_commit_cmd_domain_class(cls):
+        return None
 
     @classmethod
     def _get_external_id_relationships(cls, item):
+        exploration_ids = []
+        collection_ids = []
+
+        try:
+            for obj in item.activity_references:
+                if obj['type'] == constants.ACTIVITY_TYPE_EXPLORATION:
+                    exploration_ids.append(obj['id'])
+                elif obj['type'] == constants.ACTIVITY_TYPE_COLLECTION:
+                    collection_ids.append(obj['id'])
+        except Exception as e:
+            cls.errors['fetch properties'] = (
+                'Model id %s: Model properties cannot be fetched completely '
+                'with the error %s') % (item.id, e)
+            return {}
+
         return {
-            'activity_ids': (exp_models.ExplorationModel, item.activity_ids),
+            'exploration_ids': (exp_models.ExplorationModel, exploration_ids),
```

Comment:
We are returning 'exploration_ids' at some places and 'exploration_model' at others. Use a consistent naming for external models and please add a comment here explaining the reasoning behind the name you go with.

I feel ids sounds better than models since we are expecting external models to exist based on the ids.

Reply:
I agree with you here - ids sounds better. I have changed models to ids.